docs: update angular guidance (#84363)
grafana/grafana: docs/sources/developers/angular_deprecation/_index.md

# Angular support deprecation

> Angular plugin support is deprecated and will be removed in a future release. There are legacy Core Grafana visualizations and external plugins that rely on Grafana's Angular plugin support to work. The same is true for many internal (private) plugins that have been developed by Grafana users over the years. From Grafana v9 and onwards, there is a [server configuration option](<(https://github.com/grafana/grafana/blob/d61bcdf4ca5e69489e0067c56fbe7f0bfdf84ee4/conf/defaults.ini#L362)>) that is global to the entire instance and controls whether Angular plugin support is available or not. By default, Angular support is still enabled, but that will change soon once we complete the migration of all Angular code in the core product.

Reviewer: Sorry that I've reviewed the whole section, this is partly because GitHub's line based diffing algorithm can't give a more accurate diff.

[Semantic line breaks](https://sembr.org/) are better for future line based diffing in the GitHub UI and result in the same HTML output.

The suggestion:
- Uses semantic line breaks to improve future diffing
- Removes capitalization of "Core" because I'm not sure that it's a proper noun
- Removes "(private)" parenthetical, I'm not sure that it will be meaningful to readers.
  What would a user think a "private" plugin is?

```suggestion
Angular plugin support is deprecated and will be removed in a future release.
There are legacy core Grafana visualizations and external plugins that rely on Grafana's Angular plugin support to work.
The same is true for many internal plugins that have been developed by Grafana users over the years.
From Grafana v9 and onwards, there is a [server configuration option](https://github.com/grafana/grafana/blob/d61bcdf4ca5e69489e0067c56fbe7f0bfdf84ee4/conf/defaults.ini#L362) that's global to the entire instance and controls whether Angular plugin support is available or not.
By default, Angular support is still enabled, but that will change after the migration of all Angular code in the core product.
```

Warning messages are displayed if a dashboard depends on an a panel visualization or data source which requires AngularJS as shown in the following video:

{{< youtube id="XlEVs6g8dC8" >}}

To avoid disruption:

> - Ensure that you are running the latest version of plugins by following this guide on [updating]({{< relref "../../administration/plugin-management/#update-a-plugin" >}}). Many panels and data sources have migrated from AngularJS.
> - If you are using legacy Core Grafana visualizations such as Graph or Table-old, migrate to their replacements using the provided [automatic migrations]({{< relref "./angular-plugins/#automatic-migration-of-plugins" >}}).
> - Review the [list of current Angular plugins]({{< relref "./angular-plugins/" >}}) to discover which Core and external plugins are impacted, and whether an update or alternative is required.

Reviewer: - Links directly to the table listing AngularJS plugins
- Reorders the list to link to the table first
- Reword to provide lead in for links to tasks
- Uses semantic line breaks
- Removes capitalization of "Core" because I'm not sure that it's a proper noun

 ```suggestion
- Review the [list of current Angular plugins]({{< relref "./angular-plugins#angularjs-based-plugins" >}}) to discover which core and external plugins use Angular, and whether you need to update or use an alternative.
- Ensure that you are running the latest version of plugins.
  Many panels and data sources have migrated from AngularJS.
  To update a plugin, refer to [Update a plugin]({{< relref "../../administration/plugin-management/#update-a-plugin" >}}).
- If you are using legacy Grafana visualizations such as Graph or Table (old), migrate to their replacements.
  To migrate your visualizations automatically, refer to [Automatic migration of plugins]({{< relref "./angular-plugins#automatic-migration-of-plugins" >}}).
```

Our current plan is to completely remove any remaining support for Angular plugins in version 12. Including the removal of the [angular_support_enabled](https://github.com/grafana/grafana/blob/d61bcdf4ca5e69489e0067c56fbe7f0bfdf84ee4/conf/defaults.ini#L362) configuration parameter.

## A dashboard I use is displaying a warning, what do I need to do?

> Dashboards displaying warnings that they depend on Angular means that a panel visualization or data source being used has a dependency on this legacy framework. Contact your system administrator to advise of the issue or follow the above guidance on avoiding disruption.

Reviewer: - Uses semantic line breaks.
- Rewords in a way that feels more clear to me, but see what you think.
  If you do prefer your wording, consider the rest of the changes for the second sentence.
- Avoids "above" which isn't sympathetic to screen readers
- Adds missing word "them"

```suggestion
A dashboard displays warnings when one or more panel visualizations or data sources in the dashboard have a dependency on Angular.
Contact your system administrator to advise them of the issue or follow the preceding guidance on avoiding disruption.
```
